control-plane/agents/src/bin/core/nexus/operations.rs

```diff
@@ -269,21 +269,29 @@ impl ResourceShutdownOperations for OperationGuardArc<NexusSpec> {
             .start_update(registry, &nexus_state, NexusOperation::Shutdown)
             .await?;
 
-        let result = node.shutdown_nexus(request).await;
-        if let Err(error) = result.as_ref() {
+        let mut shutdown_failed: bool = false;
+
+        if let Err(error) = node.shutdown_nexus(request).await {
             tracing::warn!(
                 %error,
                 node.id = %node_id.as_str(),
                 nexus.uuid = %self.uuid().as_str(),
                 "Ignoring failure to complete the nexus shutdown request",
             );
+            match error {
+                SvcError::NexusNotFound { .. } => {
```

Comment:
Are you sure we get this? Don't we get `GrpcRequestError` ?
If so, might want to consider using `tonic_code` from my PR, like so: https://github.com/openebs/mayastor-control-plane/blob/5646d453e1309ee2f1a9ebbeb270c21e1207cdce/control-plane/agents/src/bin/core/controller/wrapper.rs#L1046

Reply:
Added from your PR.